app/views/shops/index.html.erb

```diff
@@ -0,0 +1,32 @@
+<h1>Magasins</h1>
+<label for="display_all">Afficher les magasins fermés</label>
+<input id="display_all" type="checkbox" />
+<div>
+  <% @shops.each do |shop| %>
+    <% if shop.closed %>
+      <div class="shop closed hide">
+    <% else %>
+      <div class="shop">
+    <% end %>
+      <%= shop.name %>
+      <%= shop.address %>
+      <%= shop.tel %>
+      <% unless shop.home_url.nil? %>
+        <% unless shop.home_url.empty? %>
+          <%= link_to "Home", shop.home_url, :target => "_blank" %>
+        <% end %>
+      <% end %>
+      <% unless shop.booking_url.nil? %>
+        <% unless shop.booking_url.empty? %>
+          <%= link_to "Booking", shop.booking_url, :target => "_blank" %>
+        <% end %>
+      <% end %>
```

Comment:
Strange, I tested this syntax this morning but it failed, empty? was tested in each case and raised an error when the URL was NIL. (even if normally, when the first condition is FALSE, the second is supposed to be skipped.
I had the same problem in the model validation.

But I tried to reproduce the case now to catch the error message and it's working 
I don't know why I have a different result now. I use a new server session, but this can explain the problem with the view only, not with the model.
And I had to restart my WSL tonight (wsl --shutdown), but I don't know if this can be an explanation.

Anyway, I changed the UNLESS  conditions in model and view, it's better.